/drivers/tty/serial/ar933x_uart.c
+++ b/drivers/tty/serial/ar933x_uart.c
@@ -33,7 +33,7 @@
 #define DRIVER_NAME "ar933x-uart"
 
 #define AR933X_UART_MAX_SCALE	0xff
-#define AR933X_UART_MAX_STEP	0xffff
+#define AR933X_UART_MAX_STEP	0x3333
 
 #define AR933X_UART_MIN_BAUD	300
 #define AR933X_UART_MAX_BAUD	3000000
